src/com/gfredericks/schpec/number.clj

```diff
@@ -0,0 +1,120 @@
+         true)))
+
+(s/def ::finite
+  (s/spec finite?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
+
+(defn- bigdec-pred
+  [precision scale]
+  (fn [d]
+    (and (or (not precision)
+             (>= precision (.precision d)))
+         (or (not scale)
+             (let [d-scale (.scale d)]
+               (and (not (neg? d-scale))
+                    (>= scale d-scale)))))))
+
+(defn bigdec-in
+  "Specs a bigdec number. Options:
+
+    :precision - the number of digits in the unscaled value (default none)
+    :scale     - the number of digits to the right of the decimal (default none)
+    :min       - minimum value (inclusive, default none)
+    :max       - maximum value (inclusive, default none)
+
+   A decimal satifies this spec if its precision and scale are not greater
+   than the specified precision and scale, if given.
+
+   Note that the java math definition of precision and scale may not be the
+   same as e.g. your database. For example, -1E-75M has a precision of 1 and a
+   scale of 75. For sanest results, you should specify both, though the spec
+   does not require both."
+  [& options]
+  (let [{:keys [precision scale min max]} options
+        dec-pred (bigdec-pred precision scale)]
+    (letfn [(pred [d]
+              (and (dec-pred d)
+                   (or (not min)
+                       (>= d min))
+                   (or (not max)
+                       (>= max d))))
+            (gen []
+              (let [min (or min
+                            (and precision
+                                 (-> BigDecimal/ONE
+                                     (.movePointRight precision)
+                                     dec
+                                     .negate)))
+                    max (or max
+                            (and precision
+                                 (-> BigDecimal/ONE
+                                     (.movePointRight precision)
+                                     dec)))
+                    mc (when precision
+                         (MathContext. precision RoundingMode/HALF_UP))]
+                (letfn [(f [d]
+                          (cond-> (bigdec d)
+                            scale
+                            (.setScale scale BigDecimal/ROUND_HALF_UP)
+                            precision
+                            (.round mc)))]
+                  (gen/fmap f (gen/double* {:infinite? false
+                                            :NaN? false
+                                            :min min
+                                            :max max})))))]
+      (s/spec pred :gen gen))))
+
+(s/def :specs.number.bigdec/precision
+  pos-int?)
+
+(s/def :specs.number.bigdec/scale
+  (s/spec (fn [x] (and (int? x) (not (neg? x))))
+          :gen #(gen/large-integer* {:min 0})))
+
+(s/def :specs.number.bigdec/min
+  (s/and bigdec?
+         ::finite))
+
+(s/def :specs.number.bigdec/max
+  (s/and bigdec?
+         ::finite))
```

Comment:
I'd prefer a naming convention for these specs that's actually scoped to `com.gfredericks.schpec`; I suppose a subnamespace would work best: `:com.gfredericks.schpec.numbers.bigdec-in/max`.